tests/permissions.py

```diff
@@ -0,0 +1,1144 @@
+    ),
+    field_id('Mutation', 'rejectDashboardShare'): TestData(
+        tenant_perm=MANAGE_DASHBOARDS, resource_perm=SHARE_DASHBOARD
+    ),
+    field_id('Mutation', 'rejectShareObject'): TestData(tenant_perm=MANAGE_SHARES, resource_perm=REJECT_SHARE_OBJECT),
+    field_id('Mutation', 'removeConsumptionRoleFromEnvironment'): TestData(
+        tenant_perm=MANAGE_ENVIRONMENTS, resource_perm=REMOVE_ENVIRONMENT_CONSUMPTION_ROLE
+    ),
+    field_id('Mutation', 'removeGroupFromEnvironment'): TestData(
+        tenant_perm=MANAGE_ENVIRONMENTS, resource_perm=REMOVE_ENVIRONMENT_GROUP
+    ),
+    field_id('Mutation', 'removeGroupFromOrganization'): TestData(
+        tenant_perm=MANAGE_ORGANIZATIONS, resource_perm=REMOVE_ORGANIZATION_GROUP
+    ),
+    field_id('Mutation', 'removeShareItemFilter'): TestData(
+        tenant_perm=MANAGE_SHARES, resource_perm=APPROVE_SHARE_OBJECT
+    ),
+    field_id('Mutation', 'removeSharedItem'): TestData(tenant_perm=MANAGE_SHARES, resource_perm=REMOVE_ITEM),
+    field_id('Mutation', 'requestDashboardShare'): TestData(
+        tenant_perm=MANAGE_DASHBOARDS, resource_ignore=IgnoreReason.NOTREQUIRED
+    ),
+    field_id('Mutation', 'revokeItemsShareObject'): TestData(tenant_perm=MANAGE_SHARES, resource_perm=GET_SHARE_OBJECT),
+    field_id('Mutation', 'startDatasetProfilingRun'): TestData(
+        tenant_perm=MANAGE_DATASETS, resource_perm=PROFILE_DATASET_TABLE
+    ),
+    field_id('Mutation', 'startGlueCrawler'): TestData(tenant_perm=MANAGE_DATASETS, resource_perm=CRAWL_DATASET),
+    field_id('Mutation', 'startMaintenanceWindow'): TestData(
+        tenant_ignore=IgnoreReason.TENANT, resource_ignore=IgnoreReason.TENANT, tenant_admin_perm=True
+    ),
+    field_id('Mutation', 'startReindexCatalog'): TestData(
+        tenant_ignore=IgnoreReason.TENANT, resource_ignore=IgnoreReason.TENANT, tenant_admin_perm=True
+    ),
+    field_id('Mutation', 'startSagemakerNotebook'): TestData(
+        tenant_perm=MANAGE_NOTEBOOKS, resource_perm=UPDATE_NOTEBOOK
+    ),
+    field_id('Mutation', 'stopMaintenanceWindow'): TestData(
+        tenant_ignore=IgnoreReason.TENANT, resource_ignore=IgnoreReason.TENANT, tenant_admin_perm=True
+    ),
+    field_id('Mutation', 'stopSagemakerNotebook'): TestData(
+        tenant_perm=MANAGE_NOTEBOOKS, resource_perm=UPDATE_NOTEBOOK
+    ),
+    field_id('Mutation', 'submitShareExtension'): TestData(
+        tenant_perm=MANAGE_SHARES, resource_perm=SUBMIT_SHARE_OBJECT
+    ),
+    field_id('Mutation', 'submitShareObject'): TestData(tenant_perm=MANAGE_SHARES, resource_perm=SUBMIT_SHARE_OBJECT),
+    field_id('Mutation', 'syncDatasetTableColumns'): TestData(
+        tenant_perm=MANAGE_DATASETS, resource_perm=UPDATE_DATASET_TABLE
+    ),
+    field_id('Mutation', 'syncTables'): TestData(tenant_perm=MANAGE_DATASETS, resource_perm=SYNC_DATASET),
+    field_id('Mutation', 'upVote'): TestData(
+        tenant_ignore=IgnoreReason.APPSUPPORT, resource_ignore=IgnoreReason.NOTREQUIRED
```

Comment:
Yes I changed this after @noah-paige suggestion because it is checking the Confidentiality Classification...
```
    @staticmethod
    def paginate_active_columns_for_table(uri: str, filter=None):
        context = get_context()
        with context.db_engine.scoped_session() as session:
            table: DatasetTable = DatasetTableRepository.get_dataset_table_by_uri(session, uri)
            dataset = DatasetRepository.get_dataset_by_uri(session, table.datasetUri)
            if (
                ConfidentialityClassification.get_confidentiality_level(dataset.confidentiality)
                != ConfidentialityClassification.Unclassified.value
            ) and (dataset.SamlAdminGroupName not in context.groups and dataset.stewards not in context.groups):
                raise exceptions.UnauthorizedOperation(
                    action='LIST_DATASET_TABLE_COLUMNS',
                    message='User is not authorized to view Columns for Confidential datasets',
                )
            return DatasetColumnRepository.paginate_active_columns_for_table(session, uri, filter)
```